### DISCUSSION TITLE
Updating individual attributes like color attribute of a plot card dynamically

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1831] [Q&A] [answered]
Hi!I am currently implementing a plot card in a Wave web app dashboard. I would like to change the colour of the plot_card on certain events like when the toggle button is toggled. 
```
q.page['tab1_mstsym']=ui.plot_card(
            box=ui.box(zone='content',order=1,width='50%'),
            title='Most Traded Symbols',
            data=db_tab1_mstsym,
            plot=ui.plot([ui.mark(type='interval', x='=Symbol', y='=TradesDone', y_min=0)])
            )
```

so when the toggle button is clicked, I would like to change the color attribute through code, something like this
`plot=ui.plot([ui.mark(type='interval', x='=Symbol', y='=TradesDone', y_min=0,color='#red')])`
How can I achieve this?How do I access individual properties of elements through code?Thanks!

**Comment by vopani** [comment]
@mturoci This will be nice to support if possible. I couldn't do this without redrawing the whole `plot_card` again.

**Reply by n-srinidhi** [reply]
Thanks!Facing the same issue

**Comment by mturoci** [comment, marked as answer]
Hm.. can you try this code?

```py
from h2o_wave import main, app, Q, ui, data


@app('/')
async def serve(q: Q):
    q.page['example'] = ui.plot_card(
        box='1 1 4 5',
        title='Line',
        data=data('year value', 8, rows=[
            ('1991', 3),
            ('1992', 4),
            ('1993', 3.5),
            ('1994', 5),
            ('1995', 4.9),
            ('1996', 6),
            ('1997', 7),
            ('1998', 9),
            ('1999', 13),
        ]),
        plot=ui.plot([ui.mark(type='line', x_scale='time', x='=year', y='=value', color='$red', y_min=0)])
    )

    await q.page.save()

    q.page['example'].plot.marks[0].color = '$blue'

    await q.page.save()

```

seems to do the job for me. cc @vopani

**Reply by n-srinidhi** [reply]
Hi!This worked. When I tried this initially it didnt work cos of another bug in my code. But works now!Thanks you.

**Reply by vopani** [reply]
Ohh nice, thanks @mturoci ! I found the mistake I was making as well 👍
